### PR TITLE
[WIP] Reader: Block Rendering Testing and Development Documentation

### LINKED ISSUE #45438: Reader: Document process for developing and testing block rendering.
Many blocks [do not render very well](https://github.com/Automattic/wp-calypso/issues/45104) (if at all) in the WP.com Reader. As a first step, let's document a process of how to test, verify, and develop solutions for Reader block rendering problems. This can be a README, or maybe a new section to the existing [Reader README](https://github.com/Automattic/wp-calypso/blob/master/client/reader/README.md).

See: pbAPfg-OF-p2
Part of: 217-gh-dotcom-manage

Site | Reader
--- | ---
Screen Shot 2020-09-07 at 12 38 56 PM | Screen Shot 2020-09-07 at 12 36 51 PM

### PR DESCRIPTION
**Pull request by krymson24**
#### Changes proposed in this Pull Request

* Adds documentation at `client/reader/README.md` to test and develop rendering blocks in the Reader pages
* Specifically, there are testing instructions to follow a site, publish a post, and then test the block in your post in the two Reader views
* Documentation of files that represent the data flow for the two Reader views
* Relevant API endpoints in which these two Reader views consume data

Outstanding: Documentation for development on the API side for how we can send more rich data for the blocks used.

#### Testing instructions

* You can test the documentation out by walking through it yourself :)

Fixes #45438

### CODE REVIEW

File: client/reader/README.md [hunk 1]
```diff
@@ -9,3 +9,77 @@ These routes are served by the module:
...
+
+Reader Posts
+
+1. `client/sections.js`
+2. `client/reader/full-post/index.js`
+3. `client/reader/full-post/controller.js`
+4. `client/blocks/reader-full-post/index.jsx`
+5. `client/components/post-excerpt/index.jsx`
+
+- The last two files render the post data. It seems that the API endpoints simply return text, but
```
Comment (mmtr) [inline client/reader/README.md:69]: Good thing is that this API response seems to include the full HTML markup ([unlike the notifications API](https://github.com/Automattic/wp-calypso/pull/45529#discussion_r486284806)):

Screen Shot 2020-09-11 at 11 41 38

This means we should be able to change the block output easily by hooking into the `render_block` filter so the block content is modified in realtime and adapted to the reader (we're exploring this same approach for emails: p1599734906020900-slack-C01ARB62VBK?thread_ts=1599683684.019300&cid=C01ARB62VBK).

**Review by mmtr — APPROVED** [review]
Looking great!

Seems that like with emails and notifications we'll need to change the block output when a post is fetched in the backend context (given that Calypso components just render what the API returns).

I think we can merge this and start identifying the involved pieces in the server/API (and document the result of that exploration in a P2 post / FG page).

File: client/reader/README.md (hunk 1 above)
Comment (krymson24) [reply client/reader/README.md:69]: Nice! I documented what I found on the API side here https://automattictraining.wordpress.com/wp-admin/post.php?post=108464&action=edit

I guess where I'm not following is -- why would we want to modify the block content? An assumption I'm making here is that the HTML that we're receiving from the endpoint is exactly what we want and how the block is constructed. It seems like what may be missing is just the styling.

Something I also need to understand better is how we can deliver the styles that are required in the HTML that is passed to the client.

**Comment by krymson24** [comment]
> Looking great!
> 
> Seems that like with emails and notifications we'll need to change the block output when a post is fetched in the backend context (given that Calypso components just render what the API returns).
> 
> I think we can merge this and start identifying the involved pieces in the server/API (and document the result of that exploration in a P2 post / FG page).

Here's the FG page! PCYsg-sdq-p2

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@blackjackkent please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/888510). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Pull request merged**